core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/SegmentIndexFileStore.java

```diff
@@ -79,6 +87,44 @@ public void readAllIIndexOfSegment(String segmentPath) throws IOException {
     }
   }
 
+  /**
+   * Read all index files and keep the cache in it.
+   *
+   * @param segmentFileStore
+   * @throws IOException
+   */
+  public void readAllIIndexOfSegment(SegmentFileStore segmentFileStore, SegmentStatus status,
+      boolean ignoreStatus) throws IOException {
+    List<CarbonFile> carbonIndexFiles = new ArrayList<>();
+    if (segmentFileStore.getLocationMap() == null) {
+      return;
+    }
+    for (Map.Entry<String, SegmentFileStore.FolderDetails> locations : segmentFileStore
+        .getLocationMap().entrySet()) {
+      String location = locations.getKey();
+      if (locations.getValue().isRelative()) {
+        location =
```

Comment:
This location variable fetching and value modification can be done after the below if check

Reply:
ok